service/route53/internal/customizations/custom_error_deser_test.go

```diff
@@ -0,0 +1,122 @@
+package customizations_test
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/route53"
+	"github.com/aws/aws-sdk-go-v2/service/route53/types"
+)
+
+func TestCustomErrorDeserialization(t *testing.T) {
+	cases := map[string]struct {
+		responseStatus     int
+		responseBody       []byte
+		expectedError      string
+		expectedRequestID  string
+		expectedResponseID string
+	}{
+		"invalidChangeBatchError": {
```

Comment:
Not significant but adding a newline between the test case struct initialization would help readability. It took a little bit to see there were multiple test cases here.